src/BowlingGame.js

```diff
@@ -0,0 +1,73 @@
+/*jshint esversion: 6 */
+class BowlingGame {
+
+  constructor() {
+    this.rolls = [];
+    this.frameNumber = 1;
+  }
+
+  roll(pins) {
+    this.rolls.push(pins);
+    console.log('in the roll method');
+    this.updateFrameNumber();
+  }
+
+  score() {
+    var result = 0;
+    var rollIndex = 0;
+    for (var i = 0; i < 10; i++) {
+      if (this._isStrike(rollIndex)) {
+        result += this._strikeScore(rollIndex);
+        rollIndex += 1;
+      } else if (this._isSpare(rollIndex)) {
+        result += this._spareScore(rollIndex);
+        rollIndex += 2;
+      } else {
+        result += this._frameScore(rollIndex);
+        rollIndex += 2;
+      }
+    }
+    console.log(result);
+    return result;
+  }
+
+  updateFrameNumber() {
```

Comment:
So this is for the interface only? Given that this calculates the frame number from scratch every time it's called, maybe it could not use an instance variable, and instead just return the frameNumber.  So in the interface, you would call `game.frameNumber()` and it would be this function, that returns the variable. Otherwise, you need to be sure this function here has been called by the `roll` method before you read the value of the variable in the interface.

Reply:
Thanks!  Have amended.